alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportIndexDestinationAction.kt

```diff
@@ -0,0 +1,189 @@
+import org.elasticsearch.client.Client
+import org.elasticsearch.cluster.service.ClusterService
+import org.elasticsearch.common.inject.Inject
+import org.elasticsearch.common.settings.Settings
+import org.elasticsearch.common.xcontent.ToXContent
+import org.elasticsearch.common.xcontent.XContentFactory.jsonBuilder
+import org.elasticsearch.rest.RestRequest
+import org.elasticsearch.rest.RestStatus
+import org.elasticsearch.tasks.Task
+import org.elasticsearch.transport.TransportService
+
+private val log = LogManager.getLogger(TransportIndexDestinationAction::class.java)
+
+class TransportIndexDestinationAction @Inject constructor(
+    transportService: TransportService,
+    val client: Client,
+    actionFilters: ActionFilters,
+    val scheduledJobIndices: ScheduledJobIndices,
+    val clusterService: ClusterService,
+    settings: Settings
+) : HandledTransportAction<IndexDestinationRequest, IndexDestinationResponse>(
+        IndexDestinationAction.NAME, transportService, actionFilters, ::IndexDestinationRequest
+) {
+
+    @Volatile private var indexTimeout = AlertingSettings.INDEX_TIMEOUT.get(settings)
+
+    override fun doExecute(task: Task, request: IndexDestinationRequest, actionListener: ActionListener<IndexDestinationResponse>) {
+        IndexDestinationHandler(client, actionListener, request).start()
+    }
+
+    inner class IndexDestinationHandler(
+        private val client: Client,
+        private val actionListener: ActionListener<IndexDestinationResponse>,
+        private val request: IndexDestinationRequest
+    ) {
+
+        fun start() {
+            if (!scheduledJobIndices.scheduledJobIndexExists()) {
+                scheduledJobIndices.initScheduledJobIndex(object : ActionListener<CreateIndexResponse> {
+                    override fun onResponse(response: CreateIndexResponse) {
+                        onCreateMappingsResponse(response)
+                    }
+                    override fun onFailure(t: Exception) {
+                        actionListener.onFailure(t)
+                    }
+                })
+            } else {
+                if (!IndexUtils.scheduledJobIndexUpdated) {
+                    IndexUtils.updateIndexMapping(ScheduledJob.SCHEDULED_JOBS_INDEX, ScheduledJob.SCHEDULED_JOB_TYPE,
+                            ScheduledJobIndices.scheduledJobMappings(), clusterService.state(), client.admin().indices(),
+                            object : ActionListener<AcknowledgedResponse> {
+                                override fun onResponse(response: AcknowledgedResponse) {
+                                    onUpdateMappingsResponse(response)
+                                }
+                                override fun onFailure(t: Exception) {
+                                    actionListener.onFailure(t)
+                                }
+                            })
+                } else {
+                    prepareDestinationIndexing()
+                }
```

Comment:
Can you change the if block to `else if`, so we wont have any nested if else statements?

Reply:
Good catch, changed.